src/timeline.tsx

```diff
@@ -0,0 +1,242 @@
+        dur: '2s',
+        repeatCount: 'indefinite',
+        calcMode: 'spline',
+        keyTimes: '0; 1',
+        keySplines: '0.42, 0, 0.58, 1', // ease-in-out
+    };
+    return (
+        <circle cx="12" cy="12" r="6" strokeWidth={1} stroke={stroke} fill="none">
+            <animate attributeName="r" values="6; 11" {...animationProps} begin={begin} />
+            <animate attributeName="opacity" values="1; 0" {...animationProps} begin={begin} />
+        </circle>
+    );
+};
+
+export const TimelineItem = ({
+    asset = {kind: 'dot'},
+    state = 'inactive',
+    role = 'listitem',
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby,
+    dataAttributes,
+    children,
+}: TimelineItemProps): JSX.Element => {
+    const {isDarkMode} = useTheme();
+    const themeVariant = useThemeVariant();
+    const backgroundVariant = isDarkMode ? 'default' : themeVariant;
+    const isOverInverse = backgroundVariant === 'inverse';
+
+    const renderCompletedCircle = ({size, iconSize}: {size: number; iconSize: number}) => (
+        <Circle
+            background={
+                isOverInverse
+                    ? vars.colors.backgroundContainerBrandOverInverse
+                    : vars.colors.backgroundContainerBrand
+            }
+            size={size}
+        >
+            <IconCheckFilled size={iconSize} color={vars.colors.inverse} />
+        </Circle>
+    );
+
+    const renderAsset = () => {
+        if (!asset) {
+            return null;
+        }
+        if (typeof asset === 'object' && 'kind' in asset) {
+            const completedOrActiveColor = isOverInverse
+                ? vars.colors.controlActivatedInverse
+                : vars.colors.controlActivated;
+
+            const inactiveColor = isOverInverse ? vars.colors.controlInverse : vars.colors.control;
+
+            const bareAssetColor = {
+                inactive: inactiveColor,
+                active: completedOrActiveColor,
+                completed: completedOrActiveColor,
+            }[state];
+
+            if (asset.kind === 'dot') {
+                return (
+                    <svg width="24" height="24" viewBox="0 0 24 24">
+                        {state === 'active' && !isRunningAcceptanceTest() && (
+                            <>
+                                <AnimatedCircle stroke={bareAssetColor} />
+                                <AnimatedCircle stroke={bareAssetColor} begin="0.5s" />
+                            </>
+                        )}
+                        <circle
+                            cx="12"
+                            cy="12"
+                            r="6"
+                            strokeWidth={2}
+                            stroke={bareAssetColor}
+                            fill={state === 'completed' ? bareAssetColor : 'none'}
+                        />
+                    </svg>
+                );
+            } else if (asset.kind === 'number') {
```

Comment:
[matter of taste] if all `if` branches have its own return, I prefer to not use `else`, so all `if`s are in the same level

Reply:
I think a switch is better here, changed